server/eventlog.go

```diff
@@ -495,6 +494,32 @@ func getGrubState(hash crypto.Hash, events []*pb.Event) (*pb.GrubState, error) {
 	return &pb.GrubState{Files: files, Commands: commands}, nil
 }
 
+// verifyNullTerminatedRawData checks the digest of the data.
+// Returns nil if digest match the hash of the data or the data without the last bytes (\x00).
+// The caller needs to make sure len(data) is at least 1, and data is ended with '\x00',
+// otherwise this function will return an error.
+func verifyNullTerminatedDataDigest(hash crypto.Hash, data []byte, digest []byte) error {
+	if len(data) < 1 || data[len(data)-1] != '\x00' {
+		return errors.New("given data is not null-terminated")
+	}
+	if err := verifyDataDigest(hash, data, digest); err != nil {
+		if err := verifyDataDigest(hash, data[:len(data)-1], digest); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// verifyDataDigest checks the digest of the data.
+func verifyDataDigest(hash crypto.Hash, data []byte, digest []byte) error {
+	hasher := hash.New()
```

Comment:
why are we creating a new hasher everytime? It's faster to reuse the same hash.Hash object.

Reply:
Did that just for simplicity. But ya I will add reset() and reuse the hasher for performance.